canonical-web-and-design/ubuntu.com
# Blog product cards (round 1)

## Done

- Add dynamic blog product cards in place of Marketo RTPs.
  Eg. on blog post tagged `kubeflow`, a list of relevant cards is added to the blog post right column and one is made visible at random. This PR comes with instrumentation to assess CTR on cards.

- This PR comes with cards for the following blog posts:
  - [Kubeflow](https://ubuntu.com/blog/tag/kubeflow)
  - [Bootstack](https://ubuntu.com/blog/tag/bootstack)
  - [Kubernetes](https://ubuntu.com/blog/tag/kubernetes)
  - [Ceph](https://ubuntu.com/blog/tag/ceph)
  - [Bosch](https://ubuntu.com/blog/tag/bosch)

- This is a first step, next will be added the notion of weighting, so that blog posts with tags spanning across multiple card groups are not dependent of the order of the `ifs`.
- The list of cards is maintained by PMs and Mktg and ongoing card code maintenance will be on Mktg.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Ensure blog posts listed under:
  - [Kubeflow](https://ubuntu-com-8249.demos.haus/blog/tag/kubeflow)
  - [Bootstack](https://ubuntu-com-8249.demos.haus/blog/tag/bootstack)
  - [Kubernetes](https://ubuntu-com-8249.demos.haus/blog/tag/kubernetes)
  - [Ceph](https://ubuntu-com-8249.demos.haus/blog/tag/ceph)
  - [Bosch](https://ubuntu-com-8249.demos.haus/blog/tag/bosch)

  ...have related product cards displayed right of the article (bottom on mobile), that their display is randomized (eg. for Kubeflow) and that links are working.
